File: Tokens/PlaylistNameToken.cs
using iTunesLib;

namespace iTunesRichPresence_Rewrite.Tokens {
    public class PlaylistNameToken : IToken {
        public string DisplayName => "Playlist name";
        public string Token => "%playlist_name";
        public bool ShowInToolbox => true;
        public string GetText(IiTunes iTunes) {
            return iTunes.CurrentPlaylist.Name;
        }
    }
}

File: src/main/templating/tokens/PlaylistNameToken.ts
import { Token } from '../types';
import { CurrentSong } from '../../itunes/types';

export default class PlaylistNameToken implements Token {
    token = 'playlist_name';
    displayName = 'Playlist Name';

    render(currentSong: CurrentSong): string {
        return currentSong.Playlist.Name;
    }
}
